### PR TITLE
internal/pkg/scaffold/olm-catalog: preserve certain fields, better CR/CRD updates

### PR DESCRIPTION
**Pull request by estroz**
**Description of the change:**
- internal/pkg/scaffold/olm-catalog: preserve `spec.customresourcedefinitions.owned` elements and use unique ID's for accessing each owned CRD
- internal/util/k8sutil: more robust `GetTypeMetaFromBytes()` instead of
`GetKindFromYAML()`

**Motivation for the change:** users that define a `spec.customresourcedefinitions.owned` element currently have to enter data manually (until #1162 is merged). Manually entered data should be preserved for all `spec.customresourcedefinition.owned`s that already exist; newly added CRD's will be appended and modified by users.

### CODE REVIEW

File: internal/pkg/scaffold/olm-catalog/csv_updaters.go [hunk 1]
```diff
@@ -240,40 +241,82 @@ func (u *InstallStrategyUpdate) updateDeploymentSpecs(strat *olminstall.Strategy
...
+	versions, err := getCRDVersions(crd)
+	if err != nil {
+		return errors.Wrapf(err, "failed to get owned CRD %s versions", crd.GetName())
+	}
+	for _, ver := range versions {
+		kind := crd.Spec.Names.Kind
+		crdDesc := olmapiv1alpha1.CRDDescription{
+			Name:    crd.ObjectMeta.Name,
+			Version: ver,
+			Kind:    kind,
```
Comment (hasbro17) [inline internal/pkg/scaffold/olm-catalog/csv_updaters.go]: Nit: Move Line 257 to 261
https://github.com/operator-framework/operator-sdk/blob/089fe3c0154732d5de0ef4fc5db57d88100d292e/internal/pkg/scaffold/olm-catalog/csv_updaters.go#L257-L261

```suggestion
			Kind:    crd.Spec.Names.Kind,
```

File: internal/util/k8sutil/k8sutil.go [hunk 2]
```diff
@@ -99,3 +94,22 @@ func GetDisplayName(name string) string {
...
+func GetTypeMetaFromBytes(b []byte) (t metav1.TypeMeta, err error) {
+	u := unstructured.Unstructured{}
+	r := bytes.NewReader(b)
+	dec := yaml.NewYAMLOrJSONDecoder(r, 8)
+	// There is only one YAML doc if there are no more bytes to be read or EOF
+	// is hit.
+	if err := dec.Decode(&u); err == nil && r.Len() != 0 {
+		return t, errors.New("error getting TypeMeta from bytes: more than one manifest in b")
+	} else if err != nil && err != io.EOF {
+		return t, err
```
Comment (hasbro17) [inline internal/util/k8sutil/k8sutil.go]:
```suggestion
		return t, errors.Wrap(err, "error getting TypeMeta from bytes")
```

File: internal/pkg/scaffold/olm-catalog/csv_updaters.go (hunk 1 above)
Comment (estroz) [reply internal/pkg/scaffold/olm-catalog/csv_updaters.go]: `kind` will be used below in #1162 so I figured why not add it here.

**Pull request merged**